internal/ingress/types.go

```diff
@@ -96,6 +97,28 @@ type Backend struct {
 	UpstreamHashBy string `json:"upstream-hash-by,omitempty"`
 	// LB algorithm configuration per ingress
 	LoadBalancing string `json:"load-balance,omitempty"`
+	// Denotes if a backend is virtual and has no server
+	Virtual bool `json:"virtual"`
+	// Additional virtual backends associated with the same server and endpoints, but have a different service
+	// which will receive a weighted amount of traffic redirected to them.
+	// +optional
+	VirtualBackends []VirtualBackend `json:"virtualBackends,omitempty"`
+}
+
+// VirtualBackend describes a normal Backend that is not backed by a Server. Instead, The virtual backend
+// shares a server with a real Backend.
+// +k8s:deepcopy-gen=true
+type VirtualBackend struct {
```

Comment:
I find this name counterintuitive. This is not a backend, it is rather a construct to augment actual backend object and make it "virtual".

Reply:
A virtual backend is actually of type `Backend` where `Virtual` attribute is `true`.